src/lib/address_bitset.cc

```diff
@@ -0,0 +1,65 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0. This product includes software
+// developed at Datadog (https://www.datadoghq.com/). Copyright 2021-Present
+// Datadog, Inc.
+#include "address_bitset.hpp"
+
+#include <cassert>
+#include <functional>
+#include <unlikely.hpp>
+
+namespace ddprof {
+
+bool AddressBitset::set(uintptr_t addr) {
+  size_t hash_addr = std::hash<uintptr_t>().operator()(addr);
+  int significant_bits = hash_addr & _nb_bits_mask;
+  auto res = std::div(significant_bits, 64);
+  unsigned index_array = res.quot;
+  unsigned bit_offset = res.rem;
+  bool success = false;
+  int attempt = 0;
+  do {
+    uint64_t old_value = _address_bitset[index_array].load();
+    if (old_value & (static_cast<uint64_t>(1) << bit_offset)) {
+      // element is already set (collisions are expected)
+      return false;
+    }
+    uint64_t new_value = old_value;
+    new_value |= static_cast<uint64_t>(1) << bit_offset;
+    success = _address_bitset[index_array].compare_exchange_weak(old_value,
```

Comment:
Better to use `std::atomic::fetch_or` instead of CAS.

Reply:
Then we have a possible race between checking the value and actually setting it.
I'll make the change and you can tell me if you are OK with that possible race condition.